nifi-nar-bundles/nifi-atlas-bundle/nifi-atlas-reporting-task/src/main/java/org/apache/nifi/atlas/provenance/StandardAnalysisContext.java

```diff
@@ -37,13 +37,15 @@
     private final NamespaceResolver namespaceResolver;
     private final ProvenanceRepository provenanceRepository;
     private final String awsS3ModelVersion;
+    private final FilesystemPathsLevel filesystemPathsLevel;
 
     public StandardAnalysisContext(NiFiFlow nifiFlow, NamespaceResolver namespaceResolver,
-                                   ProvenanceRepository provenanceRepository, String awsS3ModelVersion) {
```

Comment:
I think the arguments could be final.

Reply:
In general, I do not prefer using `final` for local variables and parameters. All these variables should be final in most cases and IDEs can mark it automatically if a variable is used in a non final way.
For existing codes I follow the existing pattern in the class/method and only add `final` if it is already used there.
This parameter list does not use `final`-s so I would not add it for the new param either.